ref/vk/r_speeds.c

```diff
@@ -706,37 +729,77 @@ static const char *getMetricTypeName(r_speeds_metric_type_t type) {
+// Actually does the job of `r_speeds_mlist` and `r_speeds_mtable` commands.
+// We can't just directly call this function from little command handler ones, because
+// all the metrics calculations happen inside `R_SpeedsDisplayMore` function.
+static void doPrintMetrics( r_speeds_mprint_mode_t *print_mode, const char *print_filter ) {
+	r_speeds_mprint_mode_t mode = *print_mode;
+	if ( mode == kSpeedsMprintNone )
+		return;
+
+	const char *header_format = NULL;
+	const char *line_format = NULL;
+	const char *row_format = NULL;
+	char line[64];
+	if ( mode == kSpeedsMprintTable ) {
+		// Note:
+		// This table alignment method relies on monospace font
+		// and will have its alignment completly broken without one.
+		// That is why there is now cvar called `r_speeds_metrics_as_table`,
+		// which can toggle this printing method on/off.
+		header_format = "  | %-38s | %-10s | %-40s | %21s\n";
+		line_format   = "  | %.38s | %.10s | %.40s | %.21s\n";
+		row_format    = "  | ^2%-38s^7 | ^3%-10s^7 | ^5%-40s^7 | ^6%s:%d^7\n";
+		memset( line, '-', sizeof( line ) );
 	} else {
-		g_speeds.frame.list_metrics_filter[0] = '\0';
+		header_format = "  %s = %s  -->  (%s, %s)\n";
+		line_format   = "";
+		row_format    = "  ^2%s^7 = ^3%s^7  -->  (^5%s^7, ^6%s:%d^7)\n";
 	}
 
-	g_speeds.frame.list_metrics = true;
-}
-
-static void doListMetrics( void ) {
-	if (!g_speeds.frame.list_metrics)
-		return;
+	// Reset mode to print only this frame.
+	*print_mode = kSpeedsMprintNone;
 
-	g_speeds.frame.list_metrics = false;
-	const char *const filter = g_speeds.frame.list_metrics_filter;
-
-	for (int i = 0; i < g_speeds.metrics_count; ++i) {
+	gEngine.Con_Printf( header_format, "module.metric_name", "value", "variable", "registration_location" );
+	gEngine.Con_Printf( line_format, line, line, line, line );
+	for ( int i = 0; i < g_speeds.metrics_count; ++i ) {
 		const r_speeds_metric_t *metric = g_speeds.metrics + i;
 
-		if (filter[0] && !Q_strstr(metric->name, filter))
+		if ( print_filter[0] && !Q_strstr( metric->name, print_filter ) )
 			continue;
 
-		char buf[16];
-		metricTypeSnprintf(buf, sizeof(buf), *metric->p_value, metric->type);
-		gEngine.Con_Printf("  ^2%s^7 %s, value = ^3%s^7 (^5%s^7, ^6%s:%d^7)\n", metric->name, getMetricTypeName(metric->type), buf, metric->var_name, metric->src_file, metric->src_line);
+		char value_with_unit[16];
+		metricTypeSnprintf( value_with_unit, sizeof( value_with_unit ), *metric->p_value, metric->type );
+		gEngine.Con_Printf( row_format, metric->name, value_with_unit, metric->var_name, metric->src_file, metric->src_line );
+	}
+	gEngine.Con_Printf( line_format, line, line, line, line );
+	gEngine.Con_Printf( header_format, "module.metric_name", "value", "variable", "registration_location" );
+}
+
+// Handles optional filter argument for `r_speeds_mlist` and `r_speeds_mtable` commands.
+static void handlePrintFilterArg() {
```

Comment:
Не собирается под линью, надо так:
```suggestion
static void handlePrintFilterArg(void) {
```